// coverage.db/turler/Mongo.js
const { Database } = require("quickmongo");

module.exports = class Mongo {

constructor(url) {
    if(!url || typeof url != "string" || !url.startsWith("mongodb")) throw new TypeError("Please enter a valid MongoDB URL.");
   this.db = new Database(url);
};

async set(key,value){
  if(!value) return console.log("Please enter a value and data ex.".red + `
  await db.set("value", "data")`.green)
 let data = await this.db.set(key,value);
 return data;
};

async push(key,value){
  if(!count) return console.log("Please enter a value and data ex.".red + `
  await db.push("value", "data")`.green)
  let data = await this.db.push(key,value);
  return data;
 };

 async add(key,count){
  if(!count) return console.log("Please enter a value and number ex.".red + `
  await db.add("value", number)`.green)
  let data = await this.db.add(key,count);
  return data;
 };

 async sub(key,count){
  if(!count) return console.log("Please enter a value and number ex.".red + `
  await db.sub("value", number)`.green)
  let data = await this.db.subtract(key,count);
  return data;
 };

 async subtract(key,count){
  if(!count) return console.log("Please enter a value and number ex.".red + `
  await db.subtract("value", number)`.green)
  let data = await this.db.subtract(key,count);
  return data;
 };

 async get(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.get("name")`.green)
  let data = await this.db.get(key);
  return data;
 };

 async fetch(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.fetch("name")`.green)
  let data = await this.db.fetch(key);
  return data;
 };

 async has(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.has("name")`.green)
  let data = await this.db.has(key);
  return data;
 };


 async delete(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.delete("name")`.green)
  let data = await this.db.delete(key);
  return data;
 };

 async deleteAll(){
  let data = await this.db.deleteAll();
  return data;
 };

 async clear(){
  let data = await this.db.deleteAll();
  return data;
 };

 async all(){
  let data = await this.db.all();
  return data;
 };

 async fetchAll(){
  let data = await this.db.all();
  return data;
 };

 async getAll(){
  let data = await this.db.all();
  return data;
 };


};
