// coverage.db/turler/SQL.js
const db = require('quick.db');

module.exports = class Sql {

constructor() {
   this.db = db;
};

set(key,value){
  if(!value) return console.log("Please enter a value and data ex.".red + `
  await db.set("value", "data")`.green)
 let data = this.db.set(key,value);
 return data;
};

push(key,value){
  if(!count) return console.log("Please enter a value and data ex.".red + `
  await db.push("value", "data")`.green)
  let data = this.db.push(key,value);
  return data;
 };

 add(key,count){
  if(!count) return console.log("Please enter a value and number ex.".red + `
  await db.add("value", number)`.green)
  let data = this.db.add(key,count);
  return data;
 };

 sub(key,count){
  if(!count) return console.log("Please enter a value and number ex.".red + `
  await db.sub("value", number)`.green)
  let data = this.db.subtract(key,count);
  return data;
 };

 subtract(key,count){
  if(!count) return console.log("Please enter a value and number ex.".red + `
  db.subtract("value", number)`.green)
  let data = this.db.subtract(key,count);
  return data;
 };

get(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.get("name")`.green)
  let data = this.db.get(key);
  return data;
 };

 fetch(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.fetch("name")`.green)
  let data = this.db.fetch(key);
  return data;
 };

 has(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.has("name")`.green)
  let data = this.db.has(key);
  return data;
 };


 delete(key){
  if(!key) return console.log("Please enter a value ex.".red + `
  await db.delete("name")`.green)
  let data = this.db.delete(key);
  return data;
 };

 deleteAll(){
  let data = this.db.deleteAll();
  return data;
 };

  clear(){
  let data = this.db.deleteAll();
  return data;
 };

 all(){
  let data = this.db.all();
  return data;
 };

 fetchAll(){
  let data = this.db.all();
  return data;
 };

 getAll(){
  let data = this.db.all();
  return data;
 };


};
